src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.Wasi.targets.in

```diff
@@ -9,4 +9,8 @@
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
     <UsingWasiRuntimeWorkload Condition="'$(UsingWasiRuntimeWorkload)' == ''">$(WasiNativeWorkloadAvailable)</UsingWasiRuntimeWorkload>
   </PropertyGroup>
+
+  <Target Name="FailBuild" BeforeTargets="ProcessFrameworkReferences;Restore;Build;Publish" Condition="'$(RuntimeIdentifier)' == 'wasi-wasm'">
+    <Error Text="The 'wasi-experimental' workload is not supported in .NET 9. Please downgrade to .NET 9 Preview 6 SDK or use .NET 10 SDK" />
```

Comment:
```suggestion
    <Error Text="The 'wasi-experimental' workload is not supported in .NET 9." />
```

None of these versions are supported. If there is additional context that we want to covey, that can be described in a markdown file somewhere.

Reply:
cc @pavelsavara